tests/src/3d/testqgs3drendering.cpp

```diff
@@ -964,5 +965,190 @@ void TestQgs3DRendering::testBillboardRendering()
   QVERIFY( renderCheck( "billboard_rendering_2", img2, 40 ) );
 }
 
+void TestQgs3DRendering::testColorRampShaderClassification()
```

Comment:
as this is a generic test for QgsColorRampShader rather than something that is directly related to 3D rendering, it would be good to move the testing code to a more appropriate part of testing code - i.e. to tests/src/core - if there is an existing test for QgsColorRampShader class, just add a new function there, if not, don't be shy to create a new auto test class... 

Another suggestion: the tests for equal interval etc. are independent from each other - best to put them in separate functions - it will be easier to understand they are completely independent.